### PR TITLE
chore: specify env tags for each branches (#3071)

### PR DESCRIPTION
**Pull request by STRRL**
### What problem does this PR solve?

cherry-pick https://github.com/chaos-mesh/chaos-mesh/pull/3071 into release-2.1

specify certain env images for release-2.1

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [x] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: hack/env-image-tag.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+DIR="$( cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd )"
+PROJECT_DIR="$(dirname "$DIR")"
+
+if [ "$#" -eq  "0" ]; then
+  echo "Usage: $0 <env-image-name>"
+  exit 1
+fi
+
+if [[ "$1" == "dev-env" || "$1" == "build-env"  ]]; then
+  docker run -i mikefarah/yq:latest ".$1.tag" < $PROJECT_DIR/env-images.yaml
```
[the remark this answers is not in the document]
Comment (STRRL) [reply hack/env-image-tag.sh:38]: @sonatype-lify help

[the remark this answers is not in the document]
Comment (STRRL) [reply hack/env-image-tag.sh:38]: @sonatype-lift help

**Comment by STRRL** [comment]
PTAL

/cc @cwen0  @Hexilee

**Review by Andrewmatilde — APPROVED** [review]
LGTM.

**Comment by Andrewmatilde** [comment]
It may be strange there BUILD_ENV & DEV_ENV but always a same env. Can we merge it? @STRRL

**Review by Hexilee — APPROVED** [review]
LGTM

**Comment by STRRL** [comment]
> It may be strange there are BUILD_ENV & DEV_ENV but they are always the same env. Can we merge it? @STRRL

I am afraid we could not use one ENV for specifying the image tags for both `ghcr.io/chaos-mesh/buid-env` and `ghcr.io/chaos-mesh/dev-env`, because they require different `IMAGE_$(NAME)_TAG` defined by `IMAGE_TEMPLATE`

**Pull request merged**